### PR TITLE
Chapter17 regions

### PR DESCRIPTION
**Pull request by HannaMichaelis**
## Summary

Contributes to Intellitect/EssentialCSharpManuscript#116

the highlights for 17.14 won't work correctly with the other regions already in place so I just removed them

### CODE REVIEW

**Comment by BenjaminMichaelis** [comment]
Documented this error in 17.14 here: IntelliTect/EssentialCSharpManuscript#314

File: src/Chapter17/Listing17.11.DefiningAnIndexOperatorWithVariableParameters.cs [hunk 1]
```diff
@@ -2,11 +2,13 @@
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter17.Listing17_11
 {
+    #region INCLUDE
     using System;
 
     public class BinaryTree<T> 
     {
```
Comment (BenjaminMichaelis) [inline src/Chapter17/Listing17.11.DefiningAnIndexOperatorWithVariableParameters.cs]:
```suggestion
```
nit: extra line

File: src/Chapter17/Listing17.12.IteratorInterfacesPattern.cs [hunk 2]
```diff
@@ -39,4 +44,5 @@ public Pair(T first, T second) : this()
         public T Second { get; } // C# 6.0 Getter-only Autoproperty
```
Comment (BenjaminMichaelis) [inline src/Chapter17/Listing17.12.IteratorInterfacesPattern.cs]:
```suggestion
```
nit: extra line

File: src/Chapter17/Listing17.20.UsingYieldReturnInAMethodThatReturnsIEnumberable.cs [hunk 3]
```diff
@@ -4,20 +4,11 @@ namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter17.Listing17_20
...
-        {
-            var game = new Pair<string>("Redskins", "Eagles");
-            foreach(string name in game.GetReverseEnumerator())
-            {
-                Console.WriteLine(name);
-            }
-        }
-    }
-
+
```
Comment (BenjaminMichaelis) [inline src/Chapter17/Listing17.20.UsingYieldReturnInAMethodThatReturnsIEnumberable.cs]:
```suggestion
```
nit: remove extra line

**Review by BenjaminMichaelis — APPROVED** [review]
Looks good @HannaMichaelis!

File: src/Chapter17/Listing17.05.AddingItemsToADictionary.cs [hunk 4]
```diff
@@ -1,33 +1,24 @@
...
     public class Program
     {
         public static void Main()
         {
-#if !PRECSHARP5
-        // C# 6.0
-        Dictionary<string, ConsoleColor> colorMap =
-            new Dictionary<string, ConsoleColor>
+        // C# 6.0 (use {"Error", consoleColor.Red} pre-C# 6.0)
+        var colorMap = new Dictionary<string, ConsoleColor>
```
Comment (Keboo) [inline src/Chapter17/Listing17.05.AddingItemsToADictionary.cs]: nit: spacing
```suggestion
            // C# 6.0 (use {"Error", consoleColor.Red} pre-C# 6.0)
            var colorMap = new Dictionary<string, ConsoleColor>
```

**Review by Keboo — APPROVED** [review]

**Pull request merged**